kratos/modeler/voxel_mesh_generator_modeler.cpp

```diff
@@ -0,0 +1,332 @@
+{
+    return Parameters(R"(
+    {
+        "type": "bounding_box",
+        "Parameters":{}
+    }  )");
+}
+
+///
+void VoxelMeshGeneratorModeler::SetupModelPart()
+{
+    ModelPart& main_model_part = ReadModelParts();
+
+    KRATOS_INFO("Modeler") << "Generating Key planes..." << std::endl;
+    KeyPlaneGenerationFactory factory;
+    auto p_key_plane_generator = factory.Create(*this, mParameters["key_plane_generator"]);
+    p_key_plane_generator->ValidateParameters();
+    p_key_plane_generator->Generate();
+
+    KRATOS_INFO("Modeler") << "Key Planes generated" << std::endl;
+
+    KRATOS_INFO("Modeler") << "Preparing Internal Data Structure" << std::endl;
+    PreparingTheInternalDataStructure();
+    KRATOS_INFO("Modeler") << "Internal Data Structure prepared" << std::endl;
+
+    if(mParameters.Has("coloring_settings_list")){
+        ApplyColoring(mParameters["coloring_settings_list"], mParameters["default_outside_color"].GetInt());
+    }
+    if(mParameters.Has("entities_generator_list")){
+        GenerateEntities(main_model_part, mParameters["entities_generator_list"]);
+    }
+    if(mParameters.Has("model_part_operations")){
+        ApplyOperations(mParameters["model_part_operations"]);
+    }
+
+    for (const auto& single_output_settings : mParameters["output_files"]){
+        const std::string type = single_output_settings["type"].GetString();
+        if( type == "vtr"){
+            mColors.WriteParaViewVTR("Coloring.vtr");
+        }
+        else if (type == "esri_ascii") {
+            mColors.PrintAllVoxelColorsToAsciiFile(single_output_settings["file_name"].GetString());
+        }
+    }
+}
+
+ModelPart& VoxelMeshGeneratorModeler::ReadModelParts(){
+    KRATOS_ERROR_IF_NOT( mParameters.Has("input_model_part_name") )
+        << "Missing \"input_model_part_name\" in VoxelMeshGeneratorModeler Parameters." << std::endl;
+
+    const std::string input_model_part_name = mParameters["input_model_part_name"].GetString();
+    mpInputModelPart = mpModel->HasModelPart(input_model_part_name)
+        ? &mpModel->GetModelPart(input_model_part_name)
+        : &mpModel->CreateModelPart(input_model_part_name);
+
+    KRATOS_ERROR_IF_NOT(mParameters.Has("mdpa_file_name")) << "mdpa_file_name not defined" << std::endl;
+
+    const std::string data_file_name =  mParameters["mdpa_file_name"].GetString();
+
+    KRATOS_INFO_IF("::[VoxelMeshGeneratorModeler]::", mEchoLevel > 0) << "Importing Cad Model from: " << data_file_name << std::endl;
+
+    // Load the mdpa
+    if(data_file_name!=""){
+        ModelPartIO(data_file_name).ReadModelPart(*mpInputModelPart);
+    }
+
+    // Create the target model
+    const std::string output_model_part_name = mParameters["output_model_part_name"].GetString();
+    ModelPart& main_model_part = mpModel->HasModelPart(output_model_part_name)
+            ? mpModel->GetModelPart(output_model_part_name)
+            : mpModel->CreateModelPart(output_model_part_name);
+    return main_model_part;
```

Comment:
Can return the ternary directly without assign to a ref.

Reply:
Agreed, but I will change it to use an existing function that does this